src/toastsimulation/ToasterSimulation.java

```diff
@@ -0,0 +1,392 @@
+                        }
+
+                    } else {
+                        bagelShape.setPosition(
+                            bagelShape.getX() + event.getDelta().getX(),
+                            bagelShape.getY() + event.getDelta().getY());
+
+                        bagelHoleShape.setCenter(bagelShape.getCenter());
+                        if(isPaul){
+                            secretImage.setCenter(bagelShape.getCenter());
+                            secretImage.setY(bagelShape.getY() + 12);
+                        }
+                    }
+                }
+            });
+    }
+
+
+
+    /**Allows the user to drag the lever under certain conditions */
+    public void animateLever() {
+        canvas.onDrag(
+            event -> {
+                Double change = event.getDelta().getY();
+                if (isLeverObject(event.getPosition()) && inSlit == true) {
+                    lever.setPosition(
+                        lever.getX(),
+                        lever.getY() + change);
+                    bagelShape.setPosition(
+                        bagelShape.getX(),
+                        bagelShape.getY() + change);
+                    bagelHoleShape.setCenter(bagelShape.getCenter());
+
+                    if (lever.getY() <= CANVAS_HEIGHT * 0.5825) {
+                        lever.setY(CANVAS_HEIGHT * 0.5825);
+                    } else if (lever.getY() >= CANVAS_HEIGHT * 0.716667) {
+                        lever.setY(CANVAS_HEIGHT * 0.716667);
+                    }
+                } else if (isLeverObject(event.getPosition())) {
+                    lever.setPosition(
+                        lever.getX(),
+                        lever.getY() + event.getDelta().getY());
+
+                    if (lever.getY() <= CANVAS_HEIGHT * 0.5825) {
+                        lever.setY(CANVAS_HEIGHT * 0.5825);
+                    } else if (lever.getY() >= CANVAS_HEIGHT * 0.716667) {
+                        lever.setY(CANVAS_HEIGHT * 0.716667);
+                    }
+                }
+            });
+
+    }
+    /**Boolean variable that checks if the lever is in 'down' position with the bread inaserted, 
+     * this ultimately is used to set the toastBread into action 
+     */
+    public boolean isLeverDownWithBread() {
+        if (inSlit && lever.getY() >= CANVAS_HEIGHT * 0.716667) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**Specifies what happens when the toaster magically accepts the bread and returns the mysteriously beter tasting hard form of it */
+    public void toastBread() {
+        if (isLeverDownWithBread() && isToasting && toastTime / 1000 != 127) {
+            if (sumTime / 1000 <= 2) {
+                Color toastColor = new Color(245, 222, 179);
+                bagelShape.setFillColor(toastColor);
+                ratings.setText("Like biting into an icecube!");
```

Comment:
Note that the only things that change between branches of the if statement here is the color and the message. How could you simplify your code to remove the repeated parts (`setFillColor`, `setText`) outside of the if statement.